repo2docker/buildpacks/legacy/__init__.py

```diff
@@ -5,130 +5,34 @@
+    This buildpack has been deprecated.
     """
-    )
-
-    def render(self):
-        """Render buildpack into a Dockerfile.
-
-        Render legacy image source (andrewosh/binder-base at a specific commit)
-        and then prependix. Render appendix (post-build commands) at the end of
-        the Dockerfile.
-
-        """
-        segments = [
-            "FROM andrewosh/binder-base@sha256:eabde24f4c55174832ed8795faa40cea62fc9e2a4a9f1ee1444f8a2e4f9710ee",
-            self.legacy_prependix,
-        ]
-        with open("Dockerfile") as f:
-            for line in f:
-                if line.strip().startswith("FROM"):
-                    break
-            segments.append(f.read())
-        segments.append(self.legacy_appendix)
-        return "\n".join(segments)
-
-    def get_build_script_files(self):
-        """
-        Dict of files to be copied to the container image for use in building.
-
-        This is copied before the `build_scripts` & `assemble_scripts` are
-        run, so can be executed from either of them.
-
-        It's a dictionary where the key is the source file path in the host
-        system, and the value is the destination file path inside the
-        container image.
-
-        This currently adds a frozen set of Python requirements to the dict
-        of files.
-
-        """
-        return {
-            "legacy/root.frozen.yml": "/tmp/root.frozen.yml",
-            "legacy/python3.frozen.yml": "/tmp/python3.frozen.yml",
-            "legacy/apt-sources.list": "/tmp/apt-sources.list",
-        }
-
-    def build(
-        self,
-        client,
-        image_spec,
-        memory_limit,
-        build_args,
-        cache_from,
-        extra_build_kwargs,
-    ):
-        """Build a legacy Docker image."""
-        with open(self.dockerfile, "w") as f:
-            f.write(self.render())
-        for env in ("root", "python3"):
-            env_file = env + ".frozen.yml"
-            src_path = os.path.join(os.path.dirname(__file__), env_file)
-            shutil.copy(src_path, env_file)
-
-        src_path = os.path.join(os.path.dirname(__file__), "apt-sources.list")
-        shutil.copy(src_path, "apt-sources.list")
-
-        return super().build(
-            client, image_spec, memory_limit, build_args, cache_from, extra_build_kwargs
-        )
 
     def detect(self):
-        """Check if current repo should be built with the Legacy BuildPack.
-        """
+        """Check if current repo should be built with the Legacy BuildPack."""
+        log = logging.getLogger("repo2docker")
         try:
             with open("Dockerfile", "r") as f:
                 for line in f:
                     if line.startswith("FROM"):
                         if "andrewosh/binder-base" in line.split("#")[0].lower():
-                            return True
+                            log.error(
+                                "The legacy buildpack was deprecated in January 2020."
```

Comment:
```suggestion
                                "The legacy buildpack was removed in January 2020."
```
I've always thought of "deprecated" as meaning "no longer recommended and has limited support, will be removed in future"

Reply:
Good point.